src/irmin/node_intf.ml

```diff
@@ -26,13 +26,16 @@ module type S = sig
   type metadata
   (** The type for node metadata. *)
 
-  type hash
-  (** The type for keys. *)
+  type contents
+  (** The type for contents keys. *)
+
+  type node
+  (** The type for node keys. *)
```

Comment:
this change could probably be upstreamed in a separate PR (while keeping `contents = node = hash` somewhere else)